src/app/components/confirmBtcTransactionComponent/inputOutputComponent.tsx

```diff
@@ -167,24 +168,40 @@ function InputOutputComponent({ address, parsedPsbt, isExpanded, onArrowClick }:
           ))}
 
           <OutputTitleText>{t('OUTPUT')}</OutputTitleText>
-          {parsedPsbt?.outputs.map((output) => (
+          {parsedPsbt?.outputs.map((output, index) => (
             <TransferDetailContainer>
-              <TransferDetailView
-                icon={OutputIcon}
-                hideAddress
-                hideCopyButton={btcAddress === output.address || ordinalsAddress === output.address}
-                amount={`${satsToBtc(new BigNumber(output ? output.amount : 0n)).toString()} BTC`}
-                address={output.address}
-              >
-                {output.address === btcAddress || output.address === ordinalsAddress ? (
-                  <TxIdContainer>
-                    <YourAddressText>(Your Address)</YourAddressText>
+              {output.outputScript ? (
+                <TransferDetailView
+                  icon={ScriptIcon}
+                  hideAddress
+                  hideCopyButton={true}
+                  amount={`${satsToBtc(new BigNumber(output ? output.amount : 0n)).toString()} BTC`}
+                  address={output.address}
+                  outputScript={output.outputScript}
+                  outputScriptIndex={index}
+                >
+                  <SubValueText>{`${t('SCRIPT_OUTPUT')} #${index}`}</SubValueText>
```

Comment:
the #number should be a counter starting from 1 for each script output. not the index of the outputs array